h2/utilities.py

```diff
@@ -317,6 +317,10 @@ def _reject_pseudo_header_fields(headers, hdr_validation_flags):
                     "Received custom pseudo-header field %s" % header[0]
                 )
 
+            if hdr_validation_flags.is_trailer:
+                raise ProtocolError(
+                    "Received pseudo-header in trailer %s" % header[0]
+                )
```

Comment:
Rather than do this check for every pseudo-header we see, it would be better to do it once, at the end of the loop. `if hdr_validation_flags.is_trailer and seen_pseudo_header_fields`.